packages/twenty-front/src/modules/settings/serverless-functions/components/tabs/SettingsServerlessFunctionTestTab.tsx

```diff
@@ -0,0 +1,101 @@
+import { H2Title, IconPlayerPlay } from 'twenty-ui';
+import { Section } from '@/ui/layout/section/components/Section';
+import {
+  ServerlessFunctionFormValues,
+  SetServerlessFunctionFormValues,
+} from '@/settings/serverless-functions/forms/useServerlessFunctionFormValues';
+import { CodeEditor } from '@/ui/input/code-editor/components/CodeEditor';
+import styled from '@emotion/styled';
+import { useEffect, useState } from 'react';
+import { CoreEditorHeader } from '@/ui/input/code-editor/components/CodeEditorHeader';
+import { Button } from '@/ui/input/button/components/Button';
+import { LightCopyIconButton } from '@/object-record/record-field/components/LightCopyIconButton';
+
+const StyledInputsContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: ${({ theme }) => theme.spacing(4)};
+`;
+
+export const SettingsServerlessFunctionTestTab = ({
+  formValues,
+  setFormValues,
+  handleExecute,
+}: {
+  formValues: ServerlessFunctionFormValues;
+  setFormValues: SetServerlessFunctionFormValues;
+  handleExecute: () => void;
+}) => {
+  const [language, setLanguage] = useState('plaintext');
+  const [resultHeight, setResultHeight] = useState(64);
+
+  const InputHeaderButton = (
+    <Button
+      title="Run Function"
+      variant="primary"
+      accent="blue"
+      size="small"
+      Icon={IconPlayerPlay}
+      onClick={handleExecute}
+    />
+  );
+
+  const InputHeader = (
+    <CoreEditorHeader title={'Input'} rightNodes={[InputHeaderButton]} />
+  );
+
+  const OutputHeaderButton = (
+    <LightCopyIconButton copyText={formValues.output || ''} />
+  );
+
+  const OutputHeader = (
+    <CoreEditorHeader title={'Output'} rightNodes={[OutputHeaderButton]} />
+  );
+
+  useEffect(() => {
```

Comment:
to check with @charlesBochet, I believe we usually not put useEffect in components directly. We create a component Effect for this purpose